### PR TITLE
Add EEPROM encapsulation

### PR DESCRIPTION
**Pull request by Lizj96**

### CODE REVIEW

File: src/Devices/EEPROM_TC.cpp [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+void EEPROM_TC::writepHSetPoint(double value) {
+  if (ph_set != EepromReadDouble(PH_ADDRESS)) {
+    EepromWriteDouble(PH_ADDRESS, ph_set);
+  }
+}
+
+double EEPROM_TC::readTempSetPoint() { 
+  // Load from EEPROM
+  ph_set = EepromReadDouble(PH_ADDRESS);
+  temp_set = EepromReadDouble(TEMP_ADDRESS);
```
Comment (jgfoster) [inline src/Devices/EEPROM_TC.cpp:30]: This line (reading the TEMP_ADDRESS) is the only code that should be in a function named `readTempSetPoint()`. The other lines should have their own methods.

File: src/Devices/EEPROM_TC.cpp [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+}
+
+void EEPROM_TC::writepHSetPoint(double value) {
+  if (ph_set != EepromReadDouble(PH_ADDRESS)) {
+    EepromWriteDouble(PH_ADDRESS, ph_set);
+  }
+}
+
+double EEPROM_TC::readTempSetPoint() { 
+  // Load from EEPROM
```
Comment (jgfoster) [inline src/Devices/EEPROM_TC.cpp]:
```
  double value = readDouble(TEMP_ADDRESS);
  if (isnan(value)) {
    return 20;
  }
  return value;
}

File: src/Devices/EEPROM_TC.h [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+public:
+  EEPROM_TC();
+  ~EEPROM_TC();
+  void writeMacAddress(byte mac[]);
+  void writepHSetPoint(double value);
+  double readTempSetPoint();
+  void doubleWrite(int address, double value);
+  double doubleRead(int address);
+
+private:
```
Comment (jgfoster) [inline src/Devices/EEPROM_TC.h:43]: add private methods for read/write double.

**Comment by jgfoster** [comment]
We can review this after https://github.com/Arduino-CI/arduino_ci/pull/184 and https://github.com/Arduino-CI/arduino_ci/pull/198 are merged. They should resolve the size cast warnings.

**Comment by prestoncarman** [comment]
@all-contributors please add @Lizj96 for code

**Pull request merged**